core/services/gateway/handlers/functions/allowlist_test.go

```diff
@@ -105,3 +119,68 @@ func TestAllowlist_UpdatePeriodically(t *testing.T) {
 		return allowlist.Allow(common.HexToAddress(addr1)) && !allowlist.Allow(common.HexToAddress(addr3))
 	}, testutils.WaitTimeout(t), time.Second).Should(gomega.BeTrue())
 }
+func TestAllowlist_UpdateFromContract(t *testing.T) {
+	t.Parallel()
+
+	t.Run("OK-iterate_over_list_of_allowed_senders", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(testutils.Context(t))
+		client := mocks.NewClient(t)
+		client.On("LatestBlockHeight", mock.Anything).Return(big.NewInt(42), nil)
+		client.On("CallContract", mock.Anything, mock.Anything, mock.Anything).Run(func(args mock.Arguments) {
+			cancel()
+		}).Return(sampleEncodedAllowlist(t), nil)
+		config := functions.OnchainAllowlistConfig{
+			ContractAddress:           common.Address{},
+			ContractVersion:           1,
+			BlockConfirmations:        1,
+			UpdateFrequencySec:        1,
+			UpdateTimeoutSec:          1,
+			StoredAllowlistBatchSize:  2,
+			OnchainAllowlistBatchSize: 2,
+		}
+
+		orm := fmocks.NewORM(t)
+		orm.On("UpsertAllowedSender", uint64(0), common.HexToAddress(addr1)).Once().Return(nil)
```

Comment:
This code is pretty messy. Can we paramatize this to avoid so much duplicate code? (ie: some sort of looping helper function to create these mock handlers?)

Reply:
agree!, I would actually like to understand better the mocked allowed list [here](https://github.com/smartcontractkit/chainlink/blob/c3b156b6bd202870eaa6ae037b25aaada11c6852/core/services/gateway/handlers/functions/allowlist_test.go#L28) I don't fully understand where the tos contract GetAllowedSendersCount gets mocked, would prefer to change the `count` that is returning